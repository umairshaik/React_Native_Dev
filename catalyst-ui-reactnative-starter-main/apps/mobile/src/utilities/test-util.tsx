import {FC, ReactElement, ReactNode} from 'react';
import {render, RenderAPI} from '@testing-library/react-native';
import {QueryClient, QueryClientProvider} from 'react-query';
import {ThemeProvider} from '../context/theme/themeContext';

const query = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});
afterEach(() => {
  query.clear();
});
const AllTheProviders: FC<{children: ReactNode}> = ({children}) => (
  <ThemeProvider>
    <QueryClientProvider client={query}>{children}</QueryClientProvider>
  </ThemeProvider>
);

const customRender = (ui: ReactElement, options?: Omit<RenderAPI, 'wrapper'>) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export * from '@testing-library/react-native';
export {customRender as render};
