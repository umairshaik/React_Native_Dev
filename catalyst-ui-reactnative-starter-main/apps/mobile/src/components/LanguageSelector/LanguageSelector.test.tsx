import {render} from '../../utilities/test-util';
import LanguageSelector from './LanguageSelector';

describe('Language Selector Component', () => {
  test('should render language selector', () => {
    const component = render(<LanguageSelector />);

    const selector = component.queryByTestId('language-selector');
    expect(selector).toBeTruthy();
  });

  test('snapshot match', () => {
    const component = render(<LanguageSelector />);

    expect(component).toMatchSnapshot();
  });
});
