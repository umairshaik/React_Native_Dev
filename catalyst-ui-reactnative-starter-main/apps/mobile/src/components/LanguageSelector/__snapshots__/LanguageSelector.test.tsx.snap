// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language Selector Component snapshot match 1`] = `
<View>
  <RNCAndroidDialogPicker
    dropdownIconColor={-14935265}
    items={
      [
        {
          "color": null,
          "enabled": true,
          "label": "English",
          "style": {
            "backgroundColor": null,
            "color": null,
          },
        },
        {
          "color": null,
          "enabled": true,
          "label": "Hindi",
          "style": {
            "backgroundColor": null,
            "color": null,
          },
        },
      ]
    }
    onSelect={[Function]}
    selected={0}
    style={
      [
        {
          "color": "#1C1B1F",
        },
        undefined,
      ]
    }
    testID="language-selector"
  />
</View>
`;
